### PR TITLE
fix: join_2 stitching

### PR DESCRIPTION
**Pull request by shawnlewis**
we weren't stitching through join2. Tricky because join2 returns an array of dictionaries where each dictionary has a key for each of two input tables. So we use a LiteralDictObjectRecorder that contains the input tables... Seems to work but needs tests.

### CODE REVIEW

File: weave/stitch.py [hunk 1]
```diff
@@ -307,6 +328,30 @@ def stitch_node_inner(
...
+        joinKey1 = subgraph_stitch(fn1, {"row": inputs[0]}, sg)
+        joinKey2 = subgraph_stitch(fn2, {"row": inputs[1]}, sg)
+        result = LiteralDictObjectRecorder(
+            node, val={alias1: inputs[0], alias2: inputs[1]}
+        )
+        # We only pass joinKey1 in as tag. I think this is ok, as stitch has already determined
+        # that everything in the tag is a necessary column (we needed those columns to produce
+        # the join key in the first place).
+        # But I'm not totally sure!
+        result.tags["joinObj"] = joinKey1
```
Comment (tssweeney) [inline weave/stitch.py:353]: I think we could do something like `opCoalesce(joinKey1, joinKey2)`... need to write tests to be sure

File: weave/stitch.py [hunk 2]
```diff
@@ -265,12 +265,33 @@ def stitch_node_inner(
...
         if isinstance(node.from_op.inputs["key"], graph.ConstNode):
-            key = _dict_utils.unescape_dots(node.from_op.inputs["key"].val)
+            path = _dict_utils.split_escaped_string(node.from_op.inputs["key"].val)
+            key = _dict_utils.unescape_dots(path[0])
             if (
                 isinstance(inputs[0], LiteralDictObjectRecorder)
                 and key in inputs[0].val
             ):
-                return inputs[0].val[key]
+                val = inputs[0].val.get(key)
```
Comment (tssweeney) [inline weave/stitch.py:274]: this is interesting as well. This might fix some edge cases in current prod

**Review by tssweeney — APPROVED** [review]
Approving to unblock Adam... agreed we need some tests though
